src/App.jsx

```diff
@@ -1,248 +1,362 @@
-                    </span>
-                  </a>
-                </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+                {query && (
+                  <span className="icon is-right">
+                    {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                    <button
+                      data-cy="ClearButton"
+                      type="button"
+                      className="delete"
+                      onClick={() => setQuery('')}
+                    />
+                  </span>
+                )}
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+              <a
+                data-cy="AllCategories"
+                className={`button is-success mr-6 ${filterByCategories.includes('All') ? '' : 'is-outlined'}`}
+                href="#/"
+                onClick={() => {
+                  setFilterByCategories(['All']);
+                }}
               >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
-
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+                All
+              </a>
+
+              {categoriesFromServer.map((category) => {
+                const {
+                  id,
+                  title,
+                } = category;
+
+                const activeClass = filterByCategories.includes(title)
+                  ? 'is-info'
+                  : '';
+
+                return (
+                  <a
+                    key={id}
+                    data-cy="Category"
+                    className={`button mr-2 my-1 ${activeClass}`}
+                    href="#/"
+                    onClick={() => setFilterByCategories(handleSelect(title))}
+                  >
+                    {title}
+                  </a>
+                );
+              })}
+            </div>
+
+            <div className="panel-block">
+              <a
+                data-cy="ResetAllButton"
+                href="#/"
+                className="button is-link is-outlined is-fullwidth"
+                onClick={() => {
+                  setFilterByOwner('All');
+                  setQuery('');
+                  setFilterByCategories(['All']);
```

Comment:
```suggestion
                  setFilterByOwner(0);
                  setQuery('');
                  setFilterByCategories([]);
```
Also would be better to seperate the reset logic in some function